packages/Chem/src/rdkit-service/rdkit-service.ts

```diff
@@ -287,47 +294,67 @@ export class RdKitService {
 
       if (query === '')
         return {matches: new BitArray(batch.length, true), fpRes: fpResult};
-      // *********** FILTERING using fingerprints
-      const patternFpUint8Length = 256;
-      const patternFpFilterBitArray = new BitArray(batch.length, false);
-      try {
-        checkEl:
-        for (let i = 0; i < batch.length; ++i) {
-          if (fpResult.fps[i]) {
-            for (let j = 0; j < patternFpUint8Length; ++j) {
-              if ((fpResult.fps[i]![j] & fpRdKit[j]) != fpRdKit[j])
-                continue checkEl;
+      let finalBitArray = new BitArray(batch.length, false);
+      if (searchType !== SubstructureSearchType.IS_SIMILAR) {
+        // *********** FILTERING using fingerprints
+        const patternFpUint8Length = 256;
+        const patternFpFilterBitArray = new BitArray(batch.length, false);
+        try {
+          checkEl:
+          for (let i = 0; i < batch.length; ++i) {
+            if (fpResult.fps[i]) {
+              for (let j = 0; j < patternFpUint8Length; ++j) {
+                const bitToCompare = searchType === SubstructureSearchType.CONTAINS ? fpResult.fps[i]![j] : fpRdKit[j];
```

Comment:
We should handle pattern fps differently in case of seraching by substructure or superstructure. In case of substructure fp of searched pattern should be a "subset" of a target fp. Otherwise the target fp should be a "subset" of a searched pattern. Not sure how to express it simplier